### PR TITLE
[FLINK-27607][tests] Migrate module flink-connector-files to JUnit5

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

Update the flink-connector-files module to AssertJ and JUnit 5 following the [JUnit 5 Migration Guide](https://docs.google.com/document/d/1514Wa_aNB9bJUen4xm5uiuXOooOJTtXqS_Jqk9KJitU/edit)

## Brief change log

Use JUnit5 and AssertJ in tests instead of JUnit4 and Hamcrest

## Verifying this change

This change is a code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: ( no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no)
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeInformationTestBase.java [hunk 1]
```diff
@@ -21,24 +21,23 @@
...
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotEquals;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Abstract test base for type information. */
-public abstract class TypeInformationTestBase<T extends TypeInformation<?>> extends TestLogger {
+@ExtendWith(TestLoggerExtension.class)
+public abstract class TypeInformationTestBase<T extends TypeInformation<?>> {
```
Comment (zentol) [inline flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeInformationTestBase.java:35]: This must be a separate commit, and ideally even a separate ticket.

Comment (snuyanzin) [reply flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeInformationTestBase.java:35]: Yes, probably you're right
I've created a separate ticket for that https://issues.apache.org/jira/browse/FLINK-27662
and a corresponding PR https://github.com/apache/flink/pull/19744

Comment (zentol) [reply flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeInformationTestBase.java:35]: #19744 looks good to me; can you rebase this PR on top of the other PR? Then I can get started reviewing this one.

Comment (snuyanzin) [reply flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeInformationTestBase.java]: yes, sure, I've  rebased it

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java [hunk 2]
```diff
@@ -161,12 +160,12 @@ public void testCommitImmediately() throws Exception {
 
             harness.notifyOfCompletedCheckpoint(1);
             List<String> partitions = collect(harness);
-            Assert.assertEquals(Arrays.asList("1", "2"), partitions);
+            assertThat(partitions).isEqualTo(Arrays.asList("1", "2"));
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java:163]:
```suggestion
            assertThat(partitions).containsExactly("1", "2");
```

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java [hunk 3]
```diff
@@ -136,12 +135,12 @@ public void testFailover() throws Exception {
             harness.notifyOfCompletedCheckpoint(5);
             List<String> partitions = collect(harness);
             // should not contains partition {9}
-            Assert.assertEquals(Arrays.asList("4", "5", "6", "7", "8"), partitions);
+            assertThat(partitions).isEqualTo(Arrays.asList("4", "5", "6", "7", "8"));
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java:138]: see below

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java [hunk 4]
```diff
@@ -118,7 +117,7 @@ public void testFailover() throws Exception {
             state = harness.snapshot(3, 3);
             harness.notifyOfCompletedCheckpoint(3);
             List<String> partitions = collect(harness);
-            Assert.assertEquals(Arrays.asList("3", "4", "5"), partitions);
+            assertThat(partitions).isEqualTo(Arrays.asList("3", "4", "5"));
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java:120]: see below

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java [hunk 5]
```diff
@@ -92,7 +91,7 @@ public void testFailover() throws Exception {
             harness.processElement(row("4"), 0);
             harness.notifyOfCompletedCheckpoint(1);
             List<String> partitions = collect(harness);
-            Assert.assertEquals(Arrays.asList("1", "2"), partitions);
+            assertThat(partitions).isEqualTo(Arrays.asList("1", "2"));
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java:94]: see below

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java [hunk 6]
```diff
@@ -60,25 +58,26 @@
...
     private Path path;
 
-    @Before
-    public void before() throws IOException {
-        File file = TEMPORARY_FOLDER.newFile();
+    @BeforeEach
+    void before() throws IOException {
+        File file = tmpDir.toFile();
         file.delete();
         path = new Path(file.toURI());
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/stream/StreamingFileWriterTest.java:76]:
```suggestion
        path = new Path(tmpDir.resolve("tmp").toUri());
```
his part is a bit strange.

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/PartitionWriterTest.java [hunk 7]
```diff
@@ -23,26 +23,39 @@
...
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 /** Test for {@link PartitionWriter}s. */
-public class PartitionWriterTest {
+class PartitionWriterTest {
+
+    @TempDir private java.nio.file.Path tmpDir;
 
-    @Rule public final LegacyRowResource usesLegacyRows = LegacyRowResource.INSTANCE;
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/PartitionWriterTest.java:43]: you could keep this field and manually call the before/after methods. That's preferable imo to duplicating the contained logic (and if we ever create an extension it also points us to this test).

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/FileSystemOutputFormatTest.java [hunk 8]
```diff
@@ -147,39 +145,40 @@ public void testStaticPartition() throws Exception {
...
         Map<String, String> sortedContent = new TreeMap<>();
         content.forEach((file, s) -> sortedContent.put(file.getParentFile().getName(), s));
 
-        assertEquals(2, sortedContent.size());
-        assertEquals("a1,1\n" + "a2,2\n" + "a3,3\n", sortedContent.get("c=p1"));
-        assertEquals("a2,2\n", sortedContent.get("c=p2"));
-        assertFalse(new File(tmpFile.toURI()).exists());
+        assertThat(sortedContent).hasSize(2);
+        assertThat(sortedContent.get("c=p1")).isEqualTo("a1,1\n" + "a2,2\n" + "a3,3\n");
+        assertThat(sortedContent.get("c=p2")).isEqualTo("a2,2\n");
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/FileSystemOutputFormatTest.java:176]: this _could_ be re-written to `contains(entry(...,...), ...)`

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/FileSystemOutputFormatTest.java [hunk 9]
```diff
@@ -86,20 +86,19 @@ public void testClosingWithoutInput() throws Exception {
...
         ref.get().finalizeGlobal(1);
-        Map<File, String> content = getFileContentByPath(outputFile);
-        assertEquals(1, content.size());
-        assertEquals(
-                "a1,1,p1\n" + "a2,2,p1\n" + "a2,2,p2\n" + "a3,3,p1\n",
-                content.values().iterator().next());
+        Map<File, String> content = getFileContentByPath(outputPath);
+        assertThat(content).hasSize(1);
+        assertThat(content.values().iterator().next())
+                .isEqualTo("a1,1,p1\n" + "a2,2,p1\n" + "a2,2,p2\n" + "a3,3,p1\n");
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/FileSystemOutputFormatTest.java:101]:
```suggestion
        assertThat(content.values()).containsExactly("a1,1,p1\n" + "a2,2,p1\n" + "a2,2,p2\n" + "a3,3,p1\n");
```

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/FileSinkITBase.java [hunk 10]
```diff
@@ -47,20 +44,17 @@ public abstract class FileSinkITBase extends TestLogger {
...
     protected static final double FAILOVER_RATIO = 0.4;
 
-    @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
-
-    @Parameterized.Parameter public boolean triggerFailover;
-
-    @Parameterized.Parameters(name = "triggerFailover = {0}")
-    public static Collection<Object[]> params() {
-        return Arrays.asList(new Object[] {false}, new Object[] {true});
+    public static Stream<Boolean> params() {
```
Comment (zentol) [inline flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/FileSinkITBase.java:47]: can this be private?

**Review by zentol — CHANGES_REQUESTED** [review]

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/writer/FileSinkMigrationITCase.java [hunk 11]
```diff
@@ -58,17 +56,15 @@
...
  * Tests migrating from {@link StreamingFileSink} to {@link FileSink}. It trigger a savepoint for
  * the {@link StreamingFileSink} job and restore the {@link FileSink} job from the savepoint taken.
  */
-public class FileSinkMigrationITCase extends TestLogger {
+class FileSinkMigrationITCase {
 
-    @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
-
-    @Rule public final SharedObjects sharedObjects = SharedObjects.create();
+    @RegisterExtension final SharedObjectsExtension sharedObjects = SharedObjectsExtension.create();
```
[the remark this answers is not in the document]
Comment (snuyanzin) [reply flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/writer/FileSinkMigrationITCase.java]: done

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/FileSinkITBase.java (hunk 10 above)
Comment (snuyanzin) [reply flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/FileSinkITBase.java]: yep, done

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/FileSystemOutputFormatTest.java (hunk 8 above)
Comment (snuyanzin) [reply flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/FileSystemOutputFormatTest.java]: done

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/PartitionWriterTest.java (hunk 7 above)
Comment (snuyanzin) [reply flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/table/PartitionWriterTest.java:43]: ok, thanks, 
I've made these methods public and started to call them

File: flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/FileSinkCompactionSwitchITCase.java [hunk 12]
```diff
@@ -108,55 +104,56 @@ public class FileSinkCompactionSwitchITCase extends TestLogger {
 
     protected static final int NUM_BUCKETS = 4;
 
-    @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+    @TempDir private static java.nio.file.Path tmpDir;
 
-    @Rule public final SharedObjects sharedObjects = SharedObjects.create();
+    @RegisterExtension final SharedObjectsExtension sharedObjects = SharedObjectsExtension.create();
```
[the remark this answers is not in the document]
Comment (snuyanzin) [reply flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/sink/FileSinkCompactionSwitchITCase.java]: done
